tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,344 @@
+                    org_id=module_sca_manifest_org.id,
+                    product=PRDS[prod],
+                    repo=RHEL_REPOS[rh_repo_key]['name'],
+                    reposet=RHEL_REPOS[rh_repo_key]['reposet'],
+                    releasever=release_version,
+                )
+                rh_repo = module_target_sat.api.Repository(id=repo_id).read()
+                all_repos.append(rh_repo)
+                logger.info('Syncing %s repository', rh_repo_key)
+                # sync repo
+                rh_repo.sync(timeout=1800)
+    function_leapp_cv.repository = all_repos
+    function_leapp_cv = function_leapp_cv.update(['repository'])
+    logger.info('Repos to be added to the AK: %s', all_repos)
+    # Publish, promote content view to lce
+    logger.info(f'Publish, promote cv - {function_leapp_cv.name} to lce - {module_leapp_lce.name}')
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    function_leapp_cv = function_leapp_cv.read()
+
+
+@pytest.fixture
+def custom_leapp_host(upgrade_path):
+    deploy_args = {}
+    deploy_args['deploy_rhel_version'] = upgrade_path['source_version']
+    logger.info('Creating Leapp Host - RHEL %s', deploy_args)
+    with Broker(
+        workflow='deploy-rhel',
+        host_class=ContentHost,
+        deploy_rhel_version=upgrade_path['source_version'],
+        deploy_flavor=settings.flavors.default,
+    ) as chost:
+        yield chost
+
+
+@pytest.mark.parametrize(
+    'upgrade_path',
+    [
+        # {'source_version': RHEL7_VER, 'target_version': RHEL8_VER},
+        {'source_version': RHEL8_VER, 'target_version': RHEL9_VER},
+    ],
+    ids=lambda upgrade_path: f'{upgrade_path["source_version"]}'
+    f'_to_{upgrade_path["target_version"]}',
+)
+@pytest.mark.usefixtures(
+    'leapp_sat_content',
+    'register_host_with_satellite',
+    'verify_target_repo_on_satellite',
+    'precondition_check_upgrade_and_install_leapp_tool',
+    'fix_inhibitors',
+)
+def test_leapp_upgrade_rhel(
+    module_target_sat,
+    custom_leapp_host,
+    upgrade_path,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 8eccc689-3bea-4182-84f3-c121e95d54c3
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create LCE, Create CV, add repositories to it, publish and promote CV, Create AK, etc.
+        3. Register content host with AK
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    logger.info('Running test for upgrade path %s', upgrade_path)
```

Comment:
In fact, I would like to see us using logging more. It can be a great way to debug code and to understand what is happening in a test. and it could improve the speed which we perform TFA with.

I also agree that the logger.info call in this specific case does not provide much value. Robotello now logs the start of each test phase (setup, call, teardown), so there is no need to duplicate that information.

However, I think it is still a good idea to use logging in places where the results of an operation are not immediately obvious. This can help to make tests more readable and easier to debug.